orchestration/_tests/test_transfer_controller.py

```diff
@@ -299,53 +317,361 @@ def test_simple_transfer_controller_no_source_or_destination(mock_config832, tra
-    with patch("os.system", return_value=0) as mock_os_system:
-        controller = SimpleTransferController(mock_config832)
-        result = controller.copy(
-            file_path="some_dir/test_file.txt",
-            source=mock_file_system_endpoint,
-            destination=mock_file_system_endpoint,
-        )
 
-        assert result is True, "Expected True when os.system returns 0."
-        mock_os_system.assert_called_once()
-        command_called = mock_os_system.call_args[0][0]
-        assert "cp -r" in command_called, "Expected cp command in os.system call."
+    # Create real directory structure
+    source_dir = tmp_path / "source"
+    dest_dir = tmp_path / "destination"
+    source_dir.mkdir()
+    dest_dir.mkdir()
 
+    # Create actual source file
+    source_file = source_dir / "experiment" / "data.txt"
+    source_file.parent.mkdir(parents=True)
+    source_file.write_text("test content")
 
-def test_simple_transfer_controller_copy_failure(
-    mock_config832, mock_file_system_endpoint, transfer_controller_module
+    # Setup endpoints with real paths
+    source_endpoint = Mock()
+    source_endpoint.name = "source_storage"
+    source_endpoint.root_path = str(source_dir)
+
+    dest_endpoint = Mock()
+    dest_endpoint.name = "dest_storage"
+    dest_endpoint.root_path = str(dest_dir)
+
+    controller = SimpleTransferController(mock_config832)
+    result = controller.copy(
+        file_path="experiment/data.txt",
+        source=source_endpoint,
+        destination=dest_endpoint,
+    )
+
+    # Verify the result and actual file operations
+    assert result is True
+
+    # Check that the file actually exists at destination
+    dest_file = dest_dir / "experiment" / "data.txt"
+    assert dest_file.exists(), "File should be copied to destination"
+    assert dest_file.read_text() == "test content", "File content should match"
+
+
+# def test_simple_transfer_controller_copy_failure(
+#     mock_config832, mock_file_system_endpoint, transfer_controller_module
+# ):
+#     SimpleTransferController = transfer_controller_module["SimpleTransferController"]
+#     with patch("orchestration.transfer_controller.os.path.exists", return_value=True):  # ensure source file exists
+#         with patch("orchestration.transfer_controller.os.system", return_value=1) as mock_os_system:
+#             controller = SimpleTransferController(mock_config832)
+#             result = controller.copy(
+#                 file_path="some_dir/test_file.txt",
+#                 source=mock_file_system_endpoint,
+#                 destination=mock_file_system_endpoint,
+#             )
+#             assert result is False, "Expected False when os.system returns non-zero."
+#             mock_os_system.assert_called_once()
+#             command_called = mock_os_system.call_args[0][0]
+#             assert "cp -r" in command_called, "Expected cp command in os.system call."
```

Comment:
if you're happy with the new version of the test we should remove this commented out test before merging